doc/reference/trd104-syscalls.md

```diff
@@ -235,57 +245,60 @@ The 6 classes are:
 | Memop            |        5         |
 | Exit             |        6         |
 
-All of the system call classes except Yield and Exit are non-blocking. When a userspace
-process calls a Subscribe, Command, Allow, Read-Only Allow, or Memop syscall,
-the kernel will not put it on a wait queue. Instead, it will return immediately
-upon completion. The kernel scheduler may decide to suspend the process due to
-a timeslice expiration or the kernel thread being runnable, but the system calls
-themselves do not block. If an operation is long-running (e.g., I/O), its completion
-is signaled by a callback (see the Subscribe call in 4.2).
+All of the system call classes except Yield and Exit are
+non-blocking. When a userspace process calls a Subscribe, Command,
+Allow, Read-Only Allow, or Memop syscall, the kernel will not put it
+on a wait queue. Instead, it will return immediately upon
+completion. The kernel scheduler may decide to suspend the process due
+to a timeslice expiration or the kernel thread being runnable, but the
+system calls themselves do not block. If an operation is long-running
+(e.g., I/O), its completion is signaled by an upcall (see the
+Subscribe call in 4.2).
 
 Successful calls to Exit system calls do not return (the process exits).
 
-Peripheral driver-specific system calls (Subscribe, Command, Allow, Read-Only Allow)
-all include two arguments, a driver identifier and a syscall identifier. The driver identifier
-specifies which peripheral system call driver to invoke. The syscall identifier (which
-is different than the Syscall Class ID in the table above)
-specifies which instance of that system call on that driver to invoke. Both
-arguments are unsigned 32-bit integers. For example, the
-Console driver has driver identifier `0x1` and a Command to the console driver with
+System calls implemented by system call drivers (Subscribe, Command,
+Allow, Read-Only Allow) all include two arguments, a driver identifier
+and a syscall identifier. The driver identifier specifies which system
+call driver to invoke. The syscall identifier (which is different than
+the Syscall Class ID in the table above) specifies which instance of
+that system call on that driver to invoke. Both arguments are unsigned
+32-bit integers. For example, the Console system call driver has
+driver identifier `0x1` and a Command to the console driver with
```

Comment:
```suggestion
driver identifier `0x1` by default and a Command to the console driver with
```

Reply:
You can change the mappings? This seems extremely dubious. If you can change the mappings then we shouldn't have standard ones/a registry.

Reply:
I think this is a persistent confusion that ought to be clarified, but probably not here. There are several notions of what "Tock" is, at least: a kernel crate that provides a particular execution model, interfaces for capsules, low-level system call semantics (the 2.0 system call interface defined in this document), etc; also an operating system implementation that uses that kernel and provides certain basic services (a console, timers, etc), with particular standardized system call interfaces.

In the latter case, the Console driver has driver identifier `0x1`. In former case, there may or may not be a console driver at all, it might be at a different system call number, and may have a totally different interface. Maybe it's used to store beanie baby futures.

I think this document is defining the latter---the low level system call interface that the kernel enforces, not the standard set of services provided via the system call interface to boards by convention.

Also, maybe this particular use of language doesn't really matter here, as it's just an example, and the example is correct. Would using "by convention" rather than "by default" be OK with everyone (which I think is technically more precise, though at the risk of being unnecessarily verbose)?